File: backend/src/main/kotlin/com/valuedriven/nimmt/model/Game.kt
package com.valuedriven.nimmt.model

import com.valuedriven.nimmt.GameId
import com.valuedriven.nimmt.PlayerId
import java.util.*

data class Game(val id: GameId,
                val creator: PlayerId,
                val started: Boolean,
                val activePlayers: List<PlayerId>,
                val points: Map<PlayerId, List<Int>>) {
}

File: frontend/src/model/Game.ts
export type PlayerId = string

export interface Game {
    readonly id: string;
    readonly creator: string;
    readonly activePlayers: PlayerId[];
    readonly started: boolean;
}

export interface Player {
    readonly name: string;
    readonly id: PlayerId;
    readonly inGame: string | undefined;
}

export interface PlayerState {
    readonly heap: Card[]
    readonly deck: Card[]
    readonly playedCard: Card | undefined
    readonly placedCard: Boolean   
}

export interface Card {
    readonly value: number
    readonly points: number
}

export interface Row {
    readonly number: number
    readonly cards: Card[]
}